flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/StringToTimestampCastRule.java

```diff
@@ -48,12 +51,24 @@ public String generateExpression(
             String inputTerm,
             LogicalType inputLogicalType,
             LogicalType targetLogicalType) {
+        int targetPrecision = LogicalTypeChecks.getPrecision(targetLogicalType);
         if (targetLogicalType.is(LogicalTypeRoot.TIMESTAMP_WITHOUT_TIME_ZONE)) {
-            return staticCall(STRING_DATA_TO_TIMESTAMP(), inputTerm);
+            final TimestampKind targetTimestampKind = ((TimestampType) targetLogicalType).getKind();
+            if (targetTimestampKind == TimestampKind.ROWTIME
```

Comment:
Maybe this is not needed at all, I don't think we can CAST to a `ROWTIME/PROCTIME`, this can only be already set on the input field of a cast, but the target is just a type, declared as a literal by the user. @twalthr right?

Reply:
TBH independently from that, It would be nice in future to just implement `TO_TIMESTAMP` and related using directly the cast rules, so perhaps keeping this code here makes sense for that goal?

Reply:
I don't think that you can create PROCTIME/ROWTIME using TO_TIMESTAMP either.

Reply:
@matriv is right, you cannot cast to `PROCTIME/ROWTIME`, we should simplify the code here